lib/generator/baseGenerator.py

```diff
@@ -123,3 +124,47 @@ def output_runtime_result_video(self, running_time_result, suite_upload_dp):
             os.mkdir(upload_case_dp)
         if os.path.exists(self.env.converted_video_output_fp):
             shutil.move(self.env.converted_video_output_fp, upload_case_dp)
+
+    @staticmethod
+    def output_ipynb_file(global_config, index_config, output_result_dir):
+
+        # init result fp and ipynb template fp
+        input_result_fp = global_config['default-result-fn']
+        input_ipynb_template_fp = os.path.join(os.path.abspath(global_config['ipynb-template-default-dn']), index_config['ipynb-template-name'])
+        result_data = CommonUtil.load_json_file(input_result_fp)
+        converted_format_dict = {}
+
+        # store result by case_name
+        if result_data:
+            for case_full_name in result_data:
+                case_name = case_full_name.split("_", 2)[-1]
+                if case_name in converted_format_dict:
```

Comment:
not sure if this helps (might not help since we might never change this dictionary)
So, it's just a suggestion :P 

```
update_dict = {case_full_name: result_data[case_full_name]}
if case_name in converted_format_dict:
    converted_format_dict[case_name].update(update_dict)
else:
    converted_format_dict[case_name] = update_dict

```